gridscale/resource_gridscale_isoimage.go

```diff
@@ -254,14 +276,22 @@ func resourceGridscaleISOImageDelete(d *schema.ResourceData, meta interface{}) e
 	client := meta.(*gsclient.Client)
 	errorPrefix := fmt.Sprintf("delete ISO-Image (%s) resource -", d.Id())
 
-	isoimage, err := client.GetISOImage(emptyCtx, d.Id())
+	//set context with timeout when timeout is set
+	ctx := context.Background()
```

Comment:
if the timeout is not set, it will keep trying forever, because  `context.Background()` has no deadline. I think we should provide a default timeout for example 5 minutes. The user can overwrite this value by passing a timeout for the resource that it's provision time exceeds the default timeout such as PaaS resources.

Reply:
Sounds legit. We should set the default timeout instead of letting all operation run forever.